smithy-aws-traits/src/main/java/software/amazon/smithy/aws/traits/tagging/TaggableTrait.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.aws.traits.tagging;
+
+import java.util.Optional;
+import software.amazon.smithy.model.node.Node;
+import software.amazon.smithy.model.node.NodeMapper;
+import software.amazon.smithy.model.node.ObjectNode;
+import software.amazon.smithy.model.shapes.ShapeId;
+import software.amazon.smithy.model.traits.AbstractTrait;
+import software.amazon.smithy.model.traits.AbstractTraitBuilder;
+import software.amazon.smithy.model.traits.TraitService;
+import software.amazon.smithy.utils.MapUtils;
+import software.amazon.smithy.utils.SmithyBuilder;
+import software.amazon.smithy.utils.ToSmithyBuilder;
+
+/**
+ * Marks a resource shape as taggable for further model validation.
+ */
+public final class TaggableTrait extends AbstractTrait implements ToSmithyBuilder<TaggableTrait> {
+    public static final ShapeId ID = ShapeId.from("aws.api#taggable");
+    private final String property;
+    private final ShapeId tagApi;
+    private final ShapeId untagApi;
+    private final ShapeId listTagsApi;
+    private final Boolean supportsSystemTags;
+
+    private TaggableTrait(Builder builder) {
+        super(ID, builder.getSourceLocation());
+        property = builder.property;
+        tagApi = builder.tagApi;
+        untagApi = builder.untagApi;
+        listTagsApi = builder.listTagsApi;
+        supportsSystemTags = builder.supportsSystemTags;
+    }
+
+    /**
+     * Gets a boolean indicating whether or not the service supports the resource carrying system tags.
+     *
+     * @return Returns true if the service supports the resource carrying system tags.
+     *
+     */
+    public Optional<Boolean> getSupportsSystemTags() {
```

Comment:
why box this an use an Optional? I'd use the new default value stuff to set the default to `false` in the IDL and remove the boxing.